src/context/isUserAuthorized.tsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState, createContext, FC, Children, ReactNode } from "react";
+import fetch from '@/helperFunctions/fetch';
+import { toast, ToastTypes } from '@/helperFunctions/toast';
+
+const SELF_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/users/self`;
+const { ERROR } = ToastTypes;
+export const isUserAuthorizedContext = createContext<Boolean>(false);
```

Comment:
Few naming convention that is followed which is good to have in the project.
1. Context are name with Capital case IsUserAuthorizedContext , and we could go ahead with IsUserAuthorizedContextProvider which would be IsUserAuthorizedContext.Provider